tests/testthat/test-qenv_eval_code.R

```diff
@@ -91,38 +91,175 @@ testthat::test_that("an error when calling eval_code returns a qenv.error object
+
+testthat::test_that("comments fall into proper calls", {
+  # If comment is on top, it gets moved to the first call.
+  # Any other comment gets moved to the call above.
+  code <- "
+    # initial comment
+    a <- 1
+    b <- 2 # inline comment
+    c <- 3
+    # inbetween comment
+    d <- 4
+    # finishing comment
+  "
+
+  q <- eval_code(qenv(), code)
+  testthat::expect_identical(
+    get_code(q),
+    code
+  )
+})
+
+testthat::test_that("comments get pasted when they fall into calls", {
+  # If comment is on top, it gets moved to the first call.
+  # Any other comment gets moved to the call above.
+  # Comments get pasted if there are two assigned to the same call.
+  code <- "
+    # initial comment
+    a <- 1 # A comment
+    b <- 2 # inline comment
+    c <- 3 # C comment
+    # inbetween comment
+    d <- 4
+    # finishing comment
+  "
+
+  q <- eval_code(qenv(), code)
+  testthat::expect_identical(
+    get_code(q),
+    code
   )
 })
 
-testthat::test_that("eval_code with a vector of code produces one warning element per code element", {
-  q <- eval_code(qenv(), c("x <- 1", "y <- 1", "warning('warn1')"))
-  testthat::expect_equal(c("> warn1\n"), q@warnings)
+testthat::test_that("comments alone are pasted to the next/following call element", {
+  code <- c("x <- 5", "# comment", "y <- 6")
+  q <- eval_code(qenv(), code)
+  testthat::expect_identical(
+    unlist(q@code)[2],
+    paste(code[2:3], collapse = "\n")
+  )
+  testthat::expect_identical(
+    get_code(q),
+    paste(code, collapse = "\n")
+  )
 })
 
+testthat::test_that("comments at the end of src are added to the previous call element", {
+  code <- c("x <- 5", "# comment")
+  q <- eval_code(qenv(), code)
+  testthat::expect_identical(
+    unlist(q@code),
+    paste(code[1:2], collapse = "\n")
+  )
+  testthat::expect_identical(
+    get_code(q),
+    paste(code, collapse = "\n")
+  )
+})
 
-testthat::test_that("a message when calling eval_code returns a qenv object which has messages", {
-  q <- eval_code(qenv(), quote("iris_data <- head(iris)"))
-  q <- eval_code(q, quote("message('This is a message')"))
-  testthat::expect_s4_class(q, "qenv")
-  testthat::expect_equal(
-    q@messages,
-    c(
-      "",
-      "> This is a message\n"
-    )
+testthat::test_that("comments from the same line are associated with it's call", {
+  code <- c("x <- 5", " y <- 4 # comment", "z <- 5")
+  q <- eval_code(qenv(), code)
+  testthat::expect_identical(
+    unlist(q@code)[2],
+    paste0(code[2], "\n")
+  )
+  testthat::expect_identical(
+    get_code(q),
+    paste(code, collapse = "\n")
   )
 })
 
-testthat::test_that("eval_code returns a qenv object with empty messages and warnings when none are returned", {
-  q <- eval_code(qenv(), quote("iris_data <- head(iris)"))
-  testthat::expect_s4_class(q, "qenv")
-  testthat::expect_equal(q@messages, "")
-  testthat::expect_equal(q@warnings, "")
+testthat::test_that("comments alone passed to eval_code are skipped", {
```

Comment:
@gogonzo this is pretty important - comments passed alone are skipped. Unsure if we want to handle this like that